core/src/main/java/bisq/core/account/sign/SignedWitness.java

```diff
@@ -0,0 +1,156 @@
+/*
+ * This file is part of Bisq.
+ *
+ * bisq is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or (at
+ * your option) any later version.
+ *
+ * bisq is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero General Public
+ * License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with bisq. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package bisq.core.account.sign;
+
+import bisq.network.p2p.storage.P2PDataStorage;
+import bisq.network.p2p.storage.payload.CapabilityRequiringPayload;
+import bisq.network.p2p.storage.payload.DateTolerantPayload;
+import bisq.network.p2p.storage.payload.LazyProcessedPayload;
+import bisq.network.p2p.storage.payload.PersistableNetworkPayload;
+
+import bisq.common.app.Capabilities;
+import bisq.common.app.Capability;
+import bisq.common.crypto.Hash;
+import bisq.common.proto.persistable.PersistableEnvelope;
+import bisq.common.util.Utilities;
+
+import io.bisq.generated.protobuffer.PB;
+
+import com.google.protobuf.ByteString;
+
+import java.util.Date;
+import java.util.concurrent.TimeUnit;
+
+import lombok.Value;
+import lombok.extern.slf4j.Slf4j;
+
+// Supports signatures made from EC key (arbitrators) and signature created with DSA key.
+@Slf4j
+@Value
+public class SignedWitness implements LazyProcessedPayload, PersistableNetworkPayload, PersistableEnvelope,
+        DateTolerantPayload, CapabilityRequiringPayload {
+    private static final long TOLERANCE = TimeUnit.DAYS.toMillis(1);
+
+    private final boolean signedByArbitrator;
+    private final byte[] witnessHash;
+    private final byte[] signature;
+    private final byte[] signerPubKey;
+    private final byte[] witnessOwnerPubKey;
+    private final long date;
+    //TODO should we add trade amount?
```

Comment:
Is the idea to add the trade amount to add this as a weight later on as well? Of course a 1000 USD trade might not be undetected compared to a 10 USD trade.

Reply:
Not sure yet if needed. Simplicity might be better here.... but leave the comment open until 100% clear.